### ISSUE TITLE
fetch new key if old key expired

### ISSUE THREAD

**Issue by ayoyoness** [pixelated/pixelated-user-agent#851]
**Narrative:**
As a user sending an email to a recipient whose public key is expired, I want the user-agent to  re-fetch  the key of the recipient so that  my email is encrypted (and sent) using his/her new public key.

**ACs:**
When the user composes a new email to a recipient, whose key in the sender's user-agent is expired, and click the send button then the email is sent and is encrypted to the recipient's newest key. The rule is to act as if it was the first time the user sends an email to the recipient.
- it covers both cases where the recipient is from a leap-provider or just having his key in a SKS key server.
- if the recipient's newest key is also expired, then output _"message not sent: expired public keys"_ in the red error growl feedback on the browser.

This is the second step/extension of #815 
Related issue: #850

**Comment by ayoyoness** [comment]
Tasks:
- [ ] Bitmask-dev: update `keymanager.get_key` to fetch remote whenever the local one is expired
- [ ] Nickserv: investigate and fix why nickserv cannot query other nickservs
- [ ] UA: fix #625 so that KeyExpiredError (and any other exception) bubbles up and eventually be feedback to the front-end.

**Comment by ayoyoness** [comment]
quick question:
are we sure about my 2nd acceptance criteria above?
More specifically,  do we really want to tell the user: "message not sent: expired public keys"  -- for both case where either the sender or the recipient key is expired?
At the  moment, for any non-SMTPDownException, we just issue a generic "an error occurred while sending"... 

@SweetVirginia

**Comment by SweetVirginia** [comment]
@ayoyoness as a Pixelated user I would expect Pixelated to handle all this "keys things" for me. So if the sender is a Pixelated user he or she should never ever see this kind of error. Why wouldn't Pixelated renew my keys automatically? Why is it even bothering me with that? Why would it prevent me to send an email with something I can't fix by myself?

If the recipient keys are expired, it should just not encrypt and notify "This message wasn't encrypted because your contact's key expired. Pixelated will update soon new ones are available. Want to send it anyway? -sent it anyway- -save draft for later-"

What do you think?

**Comment by ayoyoness** [comment]
Yeah, I think you are right.
If it is my own key then it will just renew/extend it... and if it is recipients keys then it will send unencrypted by default anyway.

btw, we do not have that confirm modal dialog asking if you really want to send unencrypted, but that should be handled in another issue.

**Comment by bwagnerr** [comment]
@ayoyoness Hey, we got 0.9 with the latest changes again, can you check if that unblocks your issue?

**Comment by thaissiqueira** [comment]
Blocked by merge happening in #815

**Comment by zariye** [comment]
https://0xacab.org/leap/bitmask-dev/merge_requests/48
